.github/workflows/ci.yml

```diff
@@ -93,13 +56,34 @@ jobs:
           sudo rm -rf /opt/ghc
           sudo rm -rf /opt/hostedtoolcache/CodeQL
 
-      - name: Restore Kernels
+      - name: Restore Test Data
         uses: actions/cache/restore@v4
+        id: cache-test-data
         if: matrix.regression-tests
         with:
-          path: ${{ github.workspace }}/kernel
-          key: kernels-${{ env.KERNELS }}
-          fail-on-cache-miss: true
+          path: tests/regression/test_data
+          key: test-data-llvm${{ matrix.llvm }}-${{ hashFiles('tests/regression/test_specs/*') }}
+
+      # Download kernel sources:
+      # - if test_data is cached, we only need the kernel for unit tests
+      # - if test_data is not cached, we need all kernels
+      - name: Download Kernel for Unit Tests
+        if: |
+          matrix.regression-tests &&
+          steps.cache-test-data.outputs.cache-hit == 'true'
+        run:
+          nix develop .#test-kernel-buildenv --command bash -c \
+            "rhel-kernel-get 3.10.0-957.el7 --output-dir kernel"
```

Comment:
Yeah, it probably doesn't as getting one version is quite fast. Also, I plan to change the unit tests to use some upstream version as those are much quicker to get (since we're getting a tarball directly instead of extracting it from SRPM).